### ISSUE TITLE
Attaching to a 3.13 process doesn't work

### ISSUE THREAD

**Issue by rchiodo** [microsoft/debugpy#1691]
See the code here for logging debug attach parts:
https://github.com/microsoft/debugpy/blob/25955a05d87a6e3d7694f7479c1c17ded0f9af90/src/debugpy/_vendored/pydevd/pydevd_attach_to_process/windows/attach.cpp#L290

I believe the current state is that creating the 3.13 runtime during the remote thread attach is not working.

**Comment by pythoninthegrass** [comment]
Not sure if this is related, but I'm getting `TypeError: '<' not supported between instances of 'NoneType' and 'int'` similar to #1587 on v3.13.0.

My `launch.json` looks like this
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Python: Current File",
      "type": "debugpy",
      "request": "launch",
      "program": "${file}",
      "console": "integratedTerminal",
      "cwd": "${fileDirname}",
      "pythonArgs": [
        "-Xfrozen_modules=off"
      ],
      "env": {
        "PYDEVD_DISABLE_FILE_VALIDATION": "1"
      }
    }
  ]
}
```
Haven't tried 3.12, but 3.11.* works swimmingly. Toggling `PYDEVD_DISABLE_FILE_VALIDATION` has no effect.

## VSCode info
```
Version: 1.94.2
Commit: 384ff7382de624fb94dbaf6da11977bba1ecd427
Date: 2024-10-09T16:08:44.566Z (2 days ago)
Electron: 30.5.1
ElectronBuildId: 10262041
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.4.0
```
## Full traceback
```
λ  /usr/bin/env /Users/lance.stephens/git/github-audit-tool/.venv/bin/python /Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher 52695 -- /Users/lance.stephens/git/github-audit-tool/github_reporting_tool.py 
Traceback (most recent call last):
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/_pydevd_sys_monitoring/_pydevd_sys_monitoring.py", line 455, in _get_code_line_info
    return _cache[code_obj]
           ~~~~~~^^^^^^^^^^
KeyError: <code object _run_module_as_main at 0x14000a1e0, file "/Users/lance.stephens/.asdf/installs/python/3.13.0/lib/python3.13/runpy.py", line 173>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/lance.stephens/.asdf/installs/python/3.13.0/lib/python3.13/runpy.py", line 198, in _run_module_as_main
    return _run_code(code, main_globals, None,
                     "__main__", mod_spec)
  File "/Users/lance.stephens/.asdf/installs/python/3.13.0/lib/python3.13/runpy.py", line 88, in _run_code
    exec(code, run_globals)
    ~~~~^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 71, in <module>
    cli.main()
    ~~~~~~~~^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 501, in main
    run()
    ~~~^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 335, in run_file
    start_debugging(target)
    ~~~~~~~~~~~~~~~^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 323, in start_debugging
    debugpy.connect(options.address, access_token=options.adapter_access_token)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/public_api.py", line 31, in wrapper
    return wrapped(*args, **kwargs)
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/api.py", line 143, in debug
    log.reraise_exception("{0}() failed:", func.__name__, level="info")
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/api.py", line 141, in debug
    return func(address, settrace_kwargs, **kwargs)
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/api.py", line 297, in connect
    _settrace(host=host, port=port, client_access_token=access_token, **settrace_kwargs)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/api.py", line 45, in _settrace
    return pydevd.settrace(*args, **kwargs)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/pydevd.py", line 3025, in settrace
    _locked_settrace(
    ~~~~~~~~~~~~~~~~^
        host,
        ^^^^^
    ...<14 lines>...
        notify_stdin=notify_stdin,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/pydevd.py", line 3114, in _locked_settrace
    PyDevdAPI().set_dont_trace_start_end_patterns(py_db, dont_trace_start_patterns, dont_trace_end_patterns)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_api.py", line 978, in set_dont_trace_start_end_patterns
    py_db.clear_dont_trace_start_end_patterns_caches()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/pydevd.py", line 1346, in clear_dont_trace_start_end_patterns_caches
    self.on_breakpoints_changed()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/pydevd.py", line 1205, in on_breakpoints_changed
    self.set_tracing_for_untraced_contexts(breakpoints_changed=True)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/pydevd.py", line 1226, in set_tracing_for_untraced_contexts
    self.set_trace_for_frame_and_parents(thread_ident, frame)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/pydevd.py", line 2418, in set_trace_for_frame_and_parents
    pydevd_sys_monitoring.enable_code_tracing(thread_ident, frame.f_code, frame)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/_pydevd_sys_monitoring/_pydevd_sys_monitoring.py", line 701, in enable_code_tracing
    func_code_info: FuncCodeInfo = _get_func_code_info(code, frame)
                                   ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/_pydevd_sys_monitoring/_pydevd_sys_monitoring.py", line 525, in _get_func_code_info
    code_line_info = _get_code_line_info(code_obj)
  File "/Users/lance.stephens/.vscode/extensions/ms-python.debugpy-2024.11.2024092501-darwin-arm64/bundled/libs/debugpy/_vendored/pydevd/_pydevd_sys_monitoring/_pydevd_sys_monitoring.py", line 465, in _get_code_line_info
    first_line = min(line_to_offset)
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

**Comment by rchiodo** [comment]
@pythoninthegrass you're using the bits before we fixed 3.13. Try updating your python debug extension. The one with the fix should be:

Name: Python Debugger
Id: ms-python.debugpy
Description: Python Debugger extension using debugpy.
Version: 2024.11.2024101102
Publisher: Microsoft
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=ms-python.debugpy

**Comment by pythoninthegrass** [comment]
I'll give it a shot. Thanks @rchiodo !
